lib/widgets/components/tickets/buy_ticket_bottom_modal_sheet.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:coffeecard/base/strings.dart';
+import 'package:coffeecard/base/style/colors.dart';
+import 'package:coffeecard/base/style/text_styles.dart';
+import 'package:coffeecard/models/ticket/product.dart';
+import 'package:coffeecard/payment/payment_handler.dart';
+import 'package:coffeecard/widgets/components/purchase/purchase_overlay.dart';
+import 'package:coffeecard/widgets/components/tickets/bottom_modal_sheet_helper.dart';
+import 'package:coffeecard/widgets/components/tickets/rounded_button.dart';
+import 'package:flutter/material.dart';
+import 'package:gap/gap.dart';
+
+class BuyTicketBottomModalSheet extends StatelessWidget {
+  const BuyTicketBottomModalSheet({required this.product});
+
+  final Product product;
+
+  @override
+  Widget build(BuildContext context) {
+    return IntrinsicHeight(
+      // TODO: Possibily very expensive widget, look into alternatives?
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          _BottomModalSheetHelper(),
+          _ModalContent(product: product),
+        ],
+      ),
+    );
+  }
+}
+
+class _BottomModalSheetHelper extends StatelessWidget {
```

Comment:
Personally, I would prefer that this was not a seperate class, since it is relatively simple, and only user once

Reply:
It will be used more than one place in the near future (swiping a ticket).

Reply:
This is a private class in the buy_ticket_bottom_modal_sheet, not the BottomModalSheetHelper of almost the same exact name. Another reason to remove this one